src/components/esn-app-grid/esn-app-grid.tsx

```diff
@@ -0,0 +1,103 @@
+import { Component, Prop, ComponentInterface, State, Host, h } from '@stencil/core';
+import { Application } from './esn-app-grid.types';
+import { AppGridTogglerIcon } from './icons/AppGridIcon';
+import { CalendarAppIcon } from './icons/CalendarAppIcon';
+import { InboxAppIcon } from './icons/InboxAppIcon';
+import { AdminAppIcon } from './icons/AdminAppIcon';
+import { ContactsAppIcon } from './icons/ContactsAppIcon';
+import { LinShareAppIcon } from './icons/LinShareAppIcon';
+import { UnknownAppIcon } from './icons/UnknownAppIcon';
+
+const iconMapping = {
+  Admin: AdminAppIcon,
+  Calendar: CalendarAppIcon,
+  Contacts: ContactsAppIcon,
+  Inbox: InboxAppIcon,
+  LinShare: LinShareAppIcon
+};
+
+const isIconAvailable = (appName: string) => Object.keys(iconMapping).includes(appName);
+
+@Component({
+  tag: 'esn-app-grid',
+  styleUrl: 'esn-app-grid.scss',
+  shadow: true
+})
+export class EsnAppGrid implements ComponentInterface {
+  private popover: HTMLElement;
+
+  @State() applications: Application[] = [];
+
+  /**
+   * A serialized JSON object that contains information about the available apps.
+   */
+  @Prop() serializedApplications: string;
+
+  componentWillLoad() {
+    try {
+      if (!this.serializedApplications) {
+        throw new Error("The 'serializedApplications' attribute/property must be set to a serialized JSON");
+      }
+
+      const applications = JSON.parse(this.serializedApplications);
+
+      if (Array.isArray(applications) && applications.length) {
+        this.applications = applications.map(({ name, url }) => {
+          if (!isIconAvailable(name)) {
+            console.warn(
+              `The application with name ${name} does not have an icon available. A fallback icon will be used instead.`
+            );
+          }
+
+          return {
+            name,
+            url
+          };
```

Comment:
At the end you mostly use a `Array.map` to log warns

Reply:
Fixed. Now there's an util to get the app icon and log warnings when it can't find an available one.